File: leetcode2285/solution.cpp
class Solution {
public:
    long long maximumImportance(int n, vector<vector<int>>& roads) {
        /// Time: O(N log N) | Space: O(N)
        /// where N is the number of roads
        vector<long long> freqs(n, 0);

        for(vector<int> &road: roads) {
            freqs[road[0]] += 1;
            freqs[road[1]] += 1;
        }

        long long res = 0;

        sort(freqs.begin(), freqs.end());

        for(long long i = 0; i < freqs.size(); ++i) {
            res += freqs[i] * (i + 1);
        }

        return res;
    }
};

File: leetcode2285/solution.rs
impl Solution {
    pub fn maximum_importance(n: i32, roads: Vec<Vec<i32>>) -> i64 {
        /// Time O(NlogN) | Space O(N)
        /// where N is the length of roads
        let n: usize = n as usize;

        let mut freqs: Vec<i64> = vec![0; n];

        for road in roads {
            freqs[road[0] as usize] += 1;
            freqs[road[1] as usize] += 1;
        }

        freqs.sort_unstable();

        let mut res: i64 = 0;

        for i in 0..n {
            res += freqs[i] * (i + 1) as i64;
        }

        res
    }
}